charmtyles/backend/charmtyles_base.hpp

```diff
@@ -492,6 +494,30 @@ class matrix_impl : public CBase_matrix_impl
             }
 
             return;
+        case ct::util::Operation::unary_expr:
+            total_size = mat_map[node_id].rows();
+            unrolled_size = mat_map[node_id].rows() / 4;
+            remainder_start = unrolled_size * 4;
+            for (std::size_t i = 0; i != remainder_start; i += 4)
+            {
+                for (std::size_t j = 0; j != mat_map[node_id].cols(); ++j)
+                {
+                    unary_expr->operator()(i, j, mat_map[node_id](i, j));
+                    unary_expr->operator()(i, j, mat_map[node_id](i + 1, j));
```

Comment:
Shouldn't this be `unary_expr->operator()(i + 1, j, mat_map[node_id](i + 1, j))`?